runtime/vm/resolvesupport.cpp

```diff
@@ -61,7 +62,9 @@ finalFieldSetAllowed(J9VMThread *currentThread, bool isStatic, J9Method *method,
 		if (romClass->majorVersion >= 53) {
 			J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
 			J9UTF8 *name = J9ROMMETHOD_NAME(romMethod);
-			if (('<' != J9UTF8_DATA(name)[0]) || (J9UTF8_LENGTH(name) != strlen(isStatic ? "<clinit>" : "<init>"))) {
+			if (('<' != J9UTF8_DATA(name)[0])
+					|| (J9UTF8_LENGTH(name) != strlen(isStatic ? "<clinit>" : "<init>"))
+					|| ((fieldClass != callerClass) && (J9ROMCLASS_IS_UNSAFE(callerClass->romClass)))) {
```

Comment:
This needs to be `!J9ROMCLASS_IS_UNSAFE(callerClass->romClass)`.

The check should be saying "if you are different classes and not unsafe, then fail".

Reply:
A formatting nitpick:
```
if (('<' != J9UTF8_DATA(name)[0])
|| (J9UTF8_LENGTH(name) != strlen(isStatic ? "<clinit>" : "<init>"))
|| ((fieldClass != callerClass) && (J9ROMCLASS_IS_UNSAFE(callerClass->romClass)))
) {
```
This makes it easier to visually see the clauses and where they end.